common/chunkStatusLogger.go

```diff
@@ -315,6 +317,12 @@ func (csl *chunkStatusLogger) FlushLog() {
 	}
 }
 
+// CloseLogger close the chunklogger thread.
+func (csl *chunkStatusLogger) CloseLogger() {
+	csl.unsavedEntries <- nil
```

Comment:
It's safer to swap the order of these else we might have a situation where it writes to unsavedEntries channel and the main gor resumes but does not see csl.done, so falls to default and then tries to write to csl.flushDone. Since that's an unbuffered channel it'll block and the chunkStatusLogger.main gor will never exit.

Reply:
My plan is to remove channel and have flag , which we set in start , then writes csl.unsavedEntries. 
Or Instead of unbuffered done channel, create buffered channel with size 1. Then i can swap these two entries. Otherwise what i observe since its unbuffered channel when try to write to done channel it stuck as main will be at start of for loop.

Reply:
yes, let's use a buffered channel of size 1. If we use flag we have to use locking and it's not in the go spirit.
Anyways all the signalling channels should have the following attributes:
1. They should have a length equal to the number of writers, so that writer can always send the "done" signal w/o having to wait for the reader.
2. Instead of bool make them of type struct{} as that's the go recommendation for signalling channels.